pep-0702.rst

```diff
@@ -0,0 +1,278 @@
+that resolve to the deprecated overload.
+For deprecated classes and functions, this includes:
+
+* References through module, class, or instance attributes (``module.deprecated_object``,
+  ``module.SomeClass.deprecated_method``, ``module.SomeClass().deprecated_method()``)
+* Any usage of deprecated objects in their defining module
+  (``x = deprecated_object()`` in ``module.py``)
+* If ``import *`` is used, usage of deprecated objects from the
+  module (``from module import *; x = deprecated_object()``)
+* ``from`` imports (``from module import deprecated_object``)
+
+There are some additional scenarios where deprecations could come into play:
+
+* An object may implement a :data:`typing.Protocol`, but only if deprecated
+  methods are considered.
+* A class may use the ``@override`` decorator from :pep:`698` to assert that
+  its method overrides a base class method, but the base class method is
+  deprecated.
+
+As these scenarios appear complex and relatively unlikely to come up in practice,
+this PEP does not mandate that type checkers detect them.
+
+Example
+-------
+
+As an example, consider this library stub named ``library.pyi``:
+
+.. code-block:: python
+
+   from typing import deprecated
+
+   @deprecated("Use Spam instead")
+   class Ham: ...
+
+   @deprecated("It is pining for the fiords")
+   def norwegian_gray(x: int) -> int: ...
+
+   @deprecated("Only str will be allowed")
+   @overload
+   def foo(x: int) -> str: ...
+   @overload
+   def foo(x: str) -> str: ...
+
+Here is how type checkers should handle usage of this library:
+
+.. code-block:: python
+
+   from library import Ham  # error: Use of deprecated class Ham. Use Spam instead.
+
+   import library
+
+   library.norwegian_gray(1)  # error: Use of deprecated function norwegian_gray. It is pining for the fiords.
+   map(library.norwegian_gray, [1, 2, 3])  # error: Use of deprecated function norwegian_gray. It is pining for the fiords.
+
+   library.foo(1)  # error: Use of deprecated overload for foo. Only str will be allowed.
+   library.foo("x")  # no error
+
+Runtime behavior
+----------------
+
+At runtime, the decorator sets an attribute ``.__deprecated__`` on the decorated
+object. The value of the attribute is the message passed to the decorator.
+The decorator returns the original object. Notably, it does not issue a runtime
+``DeprecationWarning``.
+
+Type checker behavior
+---------------------
+
+This PEP does not specify exactly how type checkers should present deprecation
+diagnostics to their users. However, some users (e.g., application developers
+targeting only a specific version of Python) may not care about deprecations,
+while others (e.g., library developers who want their library to remain
+compatible with future versions of Python) would want to catch any use of
+deprecated functionality in their CI pipeline. Therefore, it is recommended
+that type checker provide configuration options that cover both use cases.
```

Comment:
Hopefully a deprecated use will not issue a warning/error when `# type: ignore` is added.

Reply:
Yes, the error should be `# type: ignore`able like any other type checker error. I can add an explicit mention of that if you think it's useful.

Also, unrelated typo I noticed:
```suggestion
that type checkers provide configuration options that cover both use cases.
```